### PR TITLE
Improve the config handling.

### PR DESCRIPTION
**Pull request by elopio**

### CODE REVIEW

**Comment by elopio** [comment]
I think this is the last of the possible independent refactors. The following branches will depend on each other so they can't be landed separately. This one depends on #518.

File: snapcraft/config.py [hunk 1]
```diff
@@ -13,69 +13,73 @@
...
+        self.filename = None
+        self.load()
+
+    def _section_name(self):
+        # The only section we care about is the host from the SSO url
+        url = os.environ.get('UBUNTU_SSO_API_ROOT_URL',
+                             constants.UBUNTU_SSO_API_ROOT_URL)
+        return urllib.parse.urlparse(url).netloc
+
+    def get(self, option_name):
```
Comment (sergiusens) [inline snapcraft/config.py]: does this mean people need to re-login?

**Comment by sergiusens** [comment]
fwiw it feels like `snapcraft.config` could be moved into the `storeapi`

File: snapcraft/config.py (hunk 1 above)
Comment (elopio) [inline snapcraft/config.py]: no, the config format is the same.

**Pull request merged**